src/main/java/baritone/process/ElytraProcess.java

```diff
@@ -329,4 +349,58 @@ public double placeBucketCost() {
             return COST_INF;
         }
     }
+
+    private static boolean isInBounds(BlockPos pos) {
+        return pos.getY() >= 0 && pos.getY() < 128;
+    }
+
+    private boolean isAtEdge(BlockPos pos) {
+        return ctx.world().isAirBlock(pos.north())
+                || ctx.world().isAirBlock(pos.south())
+                || ctx.world().isAirBlock(pos.east())
+                || ctx.world().isAirBlock(pos.west())
+                // corners
+                || ctx.world().isAirBlock(pos.north().west())
+                || ctx.world().isAirBlock(pos.north().east())
+                || ctx.world().isAirBlock(pos.south().west())
+                || ctx.world().isAirBlock(pos.south().east());
+    }
+
+    private boolean isSafeLandingSpot(BlockPos pos) {
+        BlockPos.MutableBlockPos mut = new BlockPos.MutableBlockPos(pos);
+        while (mut.getY() >= 0) {
+            IBlockState state = ctx.world().getBlockState(mut);
+            if (state.getMaterial().isLiquid() || state.getBlock() != Blocks.MAGMA) { // lava
+                return false;
+            }
+            if (state.getMaterial().blocksMovement()) {
+                return !isAtEdge(mut);
+            }
+            mut.setPos(mut.getX(), mut.getY() - 1, mut.getZ());
+        }
+        return false; // void
+    }
+
+    private BetterBlockPos findSafeLandingSpot() {
+        final BetterBlockPos start = ctx.playerFeet();
+        Queue<BetterBlockPos> queue = new LinkedList<>();
+        Set<BetterBlockPos> visited = new HashSet<>();
+        queue.add(start);
+
+        while (!queue.isEmpty()) {
+            BetterBlockPos pos = queue.poll();
+            if (ctx.world().isBlockLoaded(pos) && isInBounds(pos) && ctx.world().getBlockState(pos).getBlock() == Blocks.AIR) {
+                if (isSafeLandingSpot(pos)) {
+                    return pos;
+                }
+                if (visited.add(pos.north())) queue.add(pos.north());
```

Comment:
3d floodfill is fine, but, for the actual isSafeLandingSpot check, it should be slightly changed. rn it will uselessly check `(0,64,0)`, `(0,65,0)`, etc while they all are the same. there should be a new Set<BetterBlockPos> alreadyChecked. it's passed into isSafeLandingSpot and if its mutableblockpos is within that set, it exits early. the input pos is added to the set at the top of isSafeLandingSpot (it adds only the top of the column, NOT the entire column). this will work efficiently because pos.up is added to the queue before pos.down, therefore it will tend towards checking the tops of columns first.